docs/volumes.md

```diff
@@ -1,100 +1,182 @@
+
+## Background
+
+Docker also has a concept of
+[volumes](https://docs.docker.com/userguide/dockervolumes/), though it is
+somewhat looser and less managed.  In Docker, a volume is simply a directory on
+disk or in another container.  Lifetimes are not managed and until very
+recently there were only local-disk-backed volumes.  Docker now provides volume
+drivers, but the functionality is very limited, still.
+
+A Kubernetes volume, on the other hand, has an explicit lifetime - the same as
+the pod that encloses it.  When a Pod ceases to exist, the volume will cease to
+exist, too.  Perhaps more importantly than this, Kubernetes supports a number
+of types of volumes, and a Pod can use any number of them simultaneously.
+
+At its core, a volume is just a directory, possibly with some data in it, which
+is accessible to the containers in a pod.  How that directory comes to be, the
+medium that backs it, and the contents of it are determined by the particular
+volume type used.
+
+To use a volume, a pod specifies what volumes to provide for the pod (the
+[spec.volumes](http://kubernetes.io/third_party/swagger-ui/#!/v1/createPod)
+field) and where to mount those into containers(the
+[spec.containers.volumeMounts](http://kubernetes.io/third_party/swagger-ui/#!/v1/createPod)
+field).
+
+A process in a container sees a filesystem view composed from their Docker
+image and volumes.  The [Docker
+image](https://docs.docker.com/userguide/dockerimages/) is at the root of the
+filesystem hierarchy, and any volumes are mounted at the specified paths within
+the image.  Volumes can not mount onto other volumes or have hard links to
+other volumes.  Each container in the Pod must independently specify where to
+mount each volume.
 
 ## Types of Volumes
 
-Kubernetes currently supports multiple types of Volumes: emptyDir,
-gcePersistentDisk, awsElasticBlockStore, gitRepo, secret, nfs, iscsi,
-glusterfs, persistentVolumeClaim, rbd. The community welcomes additional contributions.
-
-Selected volume types are described below.
-
-### EmptyDir
-
-An EmptyDir volume is created when a Pod is bound to a Node.  It is initially empty, when the first Container command starts.  Containers in the same pod can all read and write the same files in the EmptyDir volume.  When a Pod is unbound, the data in the EmptyDir is deleted forever.
-
-Some uses for an EmptyDir are:
+Kubernetes supports several types of Volumes:
+   * emptyDir
+   * hostPath
+   * gcePersistentDisk
+   * awsElasticBlockStore
+   * nfs
+   * iscsi
+   * glusterfs
+   * rbd
+   * gitRepo
+   * secret
+   * persistentVolumeClaim
+
+We welcome additional contributions.
+
+### emptyDir
+
+An `emptyDir` volume is first created when a Pod is assigned to a Node, and
+exists as long as that Pod is running on that node.  As the name says, it is
+initially empty.  Containers in the pod can all read and write the same
+files in the `emptyDir` volume.  When a Pod is removed from a node for any
```

Comment:
"volume, though it may be mounted in different locations in different containers."